regex.go

```diff
@@ -48,13 +48,23 @@ type Regexp struct {
 }
 
 func NewRegexp(pattern string, option int) (re *Regexp, err error) {
-	re = &Regexp{pattern: pattern}
-	patternCharPtr := C.CString(pattern)
-	defer C.free(unsafe.Pointer(patternCharPtr))
+	re = &Regexp{pattern: pattern, encoding: C.ONIG_ENCODING_UTF8}
+	return initRegexp(re, option)
+}
+
+// NewRegexpASCII is equivalent of NewRegexp but matching only ASCII.
+func NewRegexpASCII(pattern string, option int) (re *Regexp, err error) {
+	re = &Regexp{pattern: pattern, encoding: C.ONIG_ENCODING_ASCII}
+	return initRegexp(re, option)
+}
 
+func initRegexp(re *Regexp, option int) (*Regexp, error) {
+	var err error
+	patternCharPtr := C.CString(re.pattern)
+	defer C.free(unsafe.Pointer(patternCharPtr))
 	mutex.Lock()
 	defer mutex.Unlock()
-	error_code := C.NewOnigRegex(patternCharPtr, C.int(len(pattern)), C.int(option), &re.regex, &re.region, &re.encoding, &re.errorInfo, &re.errorBuf)
+	error_code := C.NewOnigRegex(patternCharPtr, C.int(len(re.pattern)), C.int(option), &re.regex, &re.region, &re.encoding, &re.errorInfo, &re.errorBuf)
```

Comment:
While we're touching it anyway, I recommend we clean up the naming style: `errorCode` here and below.

Reply:
addressed in e928d1f